packages/hoppscotch-cli/src/utils/checks.ts

```diff
@@ -0,0 +1,103 @@
+import fs from "fs/promises";
+import { CommanderError } from "commander";
+import { join, extname } from "path";
+import { pipe } from "fp-ts/function";
+import {
+  translateToNewRESTCollection,
+  isHoppRESTRequest,
+} from "@hoppscotch/data";
+import * as S from "fp-ts/string";
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import { error, HoppCLIError } from "../types";
+
+/**
+ * Typeguard to check valid Hoppscotch REST Collection
+ * @param param The object to be checked
+ * @returns Boolean value corresponding to the validity check
+ */
+export function isRESTCollection(param: any): E.Either<boolean, boolean> {
```

Comment:
why does isRESTCollection return an either, it is a yes or no question. Not a failable operation. So avoid the use of Either for situations like these.

Reply:
OK.